lib/genesis/test_utils.go

```diff
@@ -98,25 +101,46 @@ func CreateTestGenesisJSONFile(t *testing.T, asRaw bool) (filename string) {
 	return filename
 }
 
-// NewTestGenesisWithTrieAndHeader generates genesis, genesis trie and genesis header
-func NewTestGenesisWithTrieAndHeader(t *testing.T) (*Genesis, *trie.Trie, *types.Header) {
-	gen, err := NewGenesisFromJSONRaw("../../chain/gssmr/genesis.json")
-	if err != nil {
-		gen, err = NewGenesisFromJSONRaw("../../../chain/gssmr/genesis.json")
+func getAbsolutePath(t *testing.T, p string) string {
+	t.Helper()
+
+	_, fullpath, _, _ := runtime.Caller(0)
+	finderPath := path.Dir(fullpath)
+
+	const sarchingFor = "go.mod"
+
+	for {
```

Comment:
Limit when you reach the root perhaps? In case `go.mod` disappears on a branch or something like that.

Reply:
Nice, I will add a requirement at the `for` beginning
```go
const boundaries = "gossamer"

for {
    require.Contains(t, finderPath, boundaries)
```

if we reach some path outside the project then the `require` fails and exit
`"/Users/.../go/src/github.com/ChainSafe" does not contain "gossamer"`